FirebaseAuth/Sources/AuthProvider/OAuth/FIROAuthProvider.m

```diff
@@ -323,9 +328,17 @@ - (void)getHeadFulLiteURLWithEventID:(NSString *)eventID
                                      }
                                      NSString *argumentsString = [strongSelf
                                          httpArgumentsStringForArgsDictionary:urlArguments];
-                                     NSString *URLString =
-                                         [NSString stringWithFormat:kHeadfulLiteURLStringFormat,
-                                                                    authDomain, argumentsString];
+                                     NSString *URLString;
+                                     if (strongSelf->_auth.requestConfiguration
+                                             .emulatorHostAndPort) {
+                                       URLString = [NSString
+                                           stringWithFormat:kHeadfulLiteEmulatorURLStringFormat,
+                                                            authDomain, argumentsString];
```

Comment:
nit: the logic here doesn't actually need `authDomain`. Consider extracting most of the `completion` block into a function that takes widget URL without parameters, and call it with either the production or emulator URL. In this way, you don't even need to call or stub `fetchAuthDomainWithRequestConfiguration` at all when emulator is used.

Reply:
IMO I think it seems neater to leave everything in the completion. Since fetchAuthDomainWithRequestConfiguration returns immediately with the emulator host, I don't think there's any downside in the current logic.